yaml/com/ibm/VPD/FRUManager.interface.yaml

```diff
@@ -0,0 +1,130 @@
+description: >
+    Implement to manage VPD in the system.
+
+methods:
+    - name: WriteKeyword
+      description: >
+          Method to update the keyword value with the given value. Updates the
+          keyword on hardware and also updates on the cache, if cache
+          information  is present in the config JSON.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Inventory/hardware path of the FRU whose keyword needs to be
+                updated.
+          - name: data
+            type:
+                variant[struct[string,string,array[byte]],
+                struct[string,array[byte]]]
+            description: >
+                Contains data required to write. If it is of IPZ type VPD the
+                data should be passed in tuple[record, keyword, value] format
+                and if it is of keyword type VPD the data will be passed in
+                tuple[keyword, data] format, etc. The format of the data will be
+                depend on the type of VPD.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: ReadKeyword
+      description: >
+          Method to read VPD keyword value. Keyword can be read from hardware or
+          from cache.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Inventory/hardware Path of the FRU whose keyword needs to be
+                read.
+          - name: data
+            type: variant[struct[string,string],struct[string]]
+            description: >
+                Contains data requied to read. If it is of IPZ type VPD the data
+                should be passed in tuple[record, keyword] format and if it is
+                of keyword type VPD the data will be passed in tuple[keyword]
+                format, etc. The format of the data will be depend on the type
+                of VPD.
+          - name: target
+            type: size
+            description: >
+                Target takes 0/1 to read keyword from cache/hardware
+                respectively.
+      returns:
+          - name: value
+            type: array[byte]
+            description: >
+                Value of the given keyword read from the given target.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: GetExpandedLocationCode
```

Comment:
is anyone using this function , Please check , also lets keep it as location code and not expanded location code, if expanded location code is not available then unexpanded location code will be returned

Reply:
yes PEL is using this function 
updated the method name